### PR TITLE
docs: Clearify event loop requirements on macOS

### LINKED ISSUE #90: MacOS Sonoma displaying more than one icon
Using MacOS Sonoma (14.0) the icon will display twice on a single monitor. Once the actual tray icon is clicked and/or the current application loses focus it disappears. 

image

### PR DESCRIPTION
**Pull request by FabianLars**
closes #90 

Soooo, i spent the whole day researching this one after finally being able to reproduce it with fullscreen terminals in the latest macos version, and it seems like winit and tao are creating/starting the event loop in some way that macos isn't toooo happy about (considering that [this](https://github.com/tane123/tray-icon-example/blob/master/examples/native.rs) works fine).
Since Winit already recommends to create Windows like this i'd say doing the same with the tray icon is not too bad.

Opening this is a draft to discuss the wording and where we should document this.

Also i don't know what to do about the redraw request i used core-foundation for. Since tray-icon is not bound to winit/tao i guess it doesn't make sense to add it to tray-icon. Is there an alternative? Does sending a user event wake it up in the same way? Would that be better for the example?

p.s. updating tao example once we decide on how to do the rest.

### CODE REVIEW

File: examples/winit.rs [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 #![allow(unused)]
 
+#[cfg(target_os = "macos")]
+use core_foundation::runloop::{CFRunLoopGetMain, CFRunLoopWakeUp};
```
Comment (amrbashir) [inline examples/winit.rs]: we can move these imports to the unsafe block below

**Review by amrbashir — COMMENTED** [review]
This looks fine for me, we can update the `tao` example and move forward with this PR

**Comment by FabianLars** [comment]
interestingly enough the core foundation usage doesn't seem to be required in tao but i added it anyway so we're on the safe side.

**Review by amrbashir — APPROVED** [review]

**Comment by tleyden** [comment]
@FabianLars Does it look feasible to backport this fix to the 1.x branch?  If so I can try to help with a PR and/or testing.

**Comment by FabianLars** [comment]
oof good question, i never looked at the old tray implemented implementation. It's in tao directly, and not in this library.

To be clear since i initially missed to explain that in the tauri issue you openened. This is _not_ fixed in Tauri yet (neither v1 nor v2). This library here is just what we use in Tauri v2 but we didn't apply the new requirements over there yet :/

**Pull request merged**